roxie/ccd/ccdserver.cpp

```diff
@@ -15102,54 +15029,15 @@ class CRoxieServerNWayInputActivity : public CRoxieServerActivity
                 if (nextIndex > numInputs)
                     throw MakeStringException(ROXIE_NWAY_INPUT_ERROR, "Index %d in RANGE selection list is out of range", nextIndex);
 
-                selectedInputs.append(inputs[nextIndex-1]);
+                newInputArray[i] = inputArray[nextIndex-1];
+                newStreamArray[i] = streamArray[nextIndex-1];
             }
+            delete [] inputArray;
+            delete [] streamArray;
+            inputArray = newInputArray;
+            streamArray = newStreamArray;
```

Comment:
I suspect this works for all current instances, but it wouldn't work if it was used in a child query instead of a graph, and the selection depended on the parent record.
The code would need to be very weird, but I could construct it.